src/binding.c

```diff
@@ -66,35 +99,38 @@ NAPI_METHOD(cpuRdtsc)
 NAPI_METHOD(cpuInfo)
 {
   napi_value exports;
-  NAPI_STATUS_THROWS_VOID(napi_create_object(env, &exports));
+  NAPI_STATUS_THROWS(napi_create_object(env, &exports));
 
   // get vendor
-  char vendor[13];
+  char vendor[CPU_VENDOR_STRING_SIZE];
   cpu_write_vendor(vendor);
-  NAPI_EXPORT_STRING(vendor);
-
+  // we can't use NAPI_EXPORT_STRING since that
+  // uses NAPI_STATUS_THROWS_VOID which means we start returning
+  // corrupt pointers on error paths.
```

Comment:
for this PR it is good explain this, but before merging it would make sense to remove this comment, because it is already explained above quite extensively.

Alternative: reference that explanation

Reply:
done